qCC/mainwindow.h

```diff
@@ -59,592 +59,2410 @@ class ccTracePolylineTool;
+	 *
+	 * This function spawns a histogram dialog with the given histogram values, minimum value, maximum value,
+	 * title, and x-axis label.
+	 *
+	 * \param histoValues The histogram values.
+	 * \param minVal The minimum value.
+	 * \param maxVal The maximum value.
+	 * \param title The title of the histogram dialog.
+	 * \param xAxisLabel The label for the x-axis of the histogram.
+	 */
+	void spawnHistogramDialog(const std::vector<unsigned> &histoValues,
+							  double minVal, double maxVal,
+							  QString title, QString xAxisLabel) override;
+
+	/**
+	 * \brief Returns the picking hub.
+	 *
+	 * This function returns the picking hub associated with the main window.
+	 *
+	 * \return A pointer to the picking hub.
+	 */
+	ccPickingHub *pickingHub() override { return m_pickingHub; }
+
+	/**
+	 * \brief Removes the specified object temporarily from the database tree.
+	 *
+	 * This function removes the given object from the database tree temporarily.
+	 * The object will not be permanently deleted, but it will be hidden from the tree view.
+	 *
+	 * \param obj A pointer to the object to be removed.
+	 */
+	ccHObjectContext removeObjectTemporarilyFromDBTree(ccHObject *obj) override;
+
+	/**
+	 * \brief Puts the specified object back into the database tree.
+	 *
+	 * This function is called to put the given object back into the database tree
+	 * using the specified context.
+	 *
+	 * \param obj The object to be put back into the database tree.
+	 * \param context The context used to put the object back into the tree.
+	 */
+	void putObjectBackIntoDBTree(ccHObject *obj, const ccHObjectContext &context) override;
+
+	/**
+	 * \brief Handles the event when an item is picked.
+	 *
+	 * This function is called when an item is picked and provides information about the picked item.
+	 *
+	 * \param pi The picked item information.
+	 */
+	void onItemPicked(const PickedItem &pi) override;
+
+	/**
+	 * \brief Returns a container of top-level selected entities.
+	 *
+	 * This function returns a container of top-level selected entities in the current context.
+	 * Top-level entities are the entities that are directly selected by the user, excluding any
+	 * entities that are children of other selected entities.
+	 *
+	 * \return A container of top-level selected entities.
+	 */
 	ccHObject::Container getTopLevelSelectedEntities() const;
 
-	//! Returns real 'dbRoot' object
-	virtual ccDBRoot* db();
+	/**
+	 * \brief Returns the ccDBRoot object.
+	 *
+	 * This function returns the ccDBRoot object, which represents the database root.
+	 *
+	 * \return A pointer to the ccDBRoot object.
+	 */
+	virtual ccDBRoot *db();
+
+	/**
+	 * \brief Adds the "Edit Plane" action to the given menu.
+	 *
+	 * This function adds an action called "Edit Plane" to the specified menu.
+	 * The action allows the user to edit a plane.
+	 *
+	 * \param menu The menu to which the action will be added.
+	 */
+	void addEditPlaneAction(QMenu &menu) const;
```

Comment:
same thing for parameters, in CloudCompare we've used "QMenu& menu" instead of "QMenu &menu"

Reply:
Equivalent to the clang option "PointerAlignment = PAS_Left"